internal/persistence/persistence.go

```diff
@@ -39,6 +44,15 @@ type Service interface {
 	CreateWorkflow(context.Context, *datatypes.Workflow) error
 
 	CreateTask(context.Context, *datatypes.Task) error
+	// CreateTasks persists all tasks yielded by the sequence. For very large
+	// sequences, the transaction may remain open for an extended period,
+	// potentially causing lock contention or timeout issues.
+	//
+	// Implementors must consume the sequence in batches and stop on the first
```

Comment:
It seem overly prescriptive to me to specify the implementation details like this. What if the back-end is a data store that doesn't allow batch inserts or support transaction locks?  I get why you've implemented the Ent/MySQL method as you have, but I don't see why a different implementation shouldn't make different choices if it leads to better performance.

Reply:
Good point, I should only state the required semantics (all-or-nothing, stop on first error...), perhaps something like the following:

```
// CreateTasks persists all tasks yielded by seq atomically: if it returns nil,
// every task was written and updated in place with generated fields (e.g. DB IDs);
// if any error occurs, no tasks are left written.
//
// The implementation should stop on the first error and must not consume seq
// more than once.
```